tests/test-verify-volume-cache.yml

```diff
@@ -0,0 +1,41 @@
+---
+
+- name: check cache options
+  block:
+
+    - name: Get information about the LV
+      command: "lvs --noheadings --nameprefixes --units=b --nosuffix --unquoted -o name,attr,cache_total_blocks,chunk_size,segtype
+                {{ storage_test_pool.name }}/{{ storage_test_volume.name }}"
+      register: lvs
+      changed_when: false
+
+    - set_fact:
+        storage_test_lv_segtype: "{{ lvs.stdout|regex_search('LVM2_SEGTYPE=(\\S+)',  '\\1') }}"
+
+    - name: check segment type
+      assert:
+        that: "{{ storage_test_lv_segtype[0] == 'cache' if storage_test_volume.cached|bool else storage_test_lv_segtype[0] == 'linear' }}"
```

Comment:
`assert` `that` values (and `when` values) do not have to use `"{{ ... }}"` - so
```
        that: storage_test_lv_segtype[0] == 'cache' if storage_test_volume.cached|bool else storage_test_lv_segtype[0] == 'linear'
```
Ansible knows that these values must be strings containing jinja2 code

Reply:
Ansible actually failed without `"{{ ... }}"`, I guess it didn't like the `if-else` construction? But I found that `ternary` exists and it looks better and works.